charts/datadog/templates/agent-clusterchecks-confd-configmap.yaml

```diff
@@ -0,0 +1,39 @@
+{{- if eq (include "need-clusterchecks-confd" .) "true" }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "datadog.fullname" . }}-clusterchecks-confd
+  namespace: {{ .Release.Namespace }}
+  labels:
+{{ include "datadog.labels" . | indent 4 }}
+data:
+{{- if .Values.datadog.kubeStateMetricsCore.enabled }}
+  kubernetes_state_core.yaml.default: |-
+    init_config:
+    instances:
+      - collectors:
+        - secrets
+        - nodes
+        - pods
+        - services
+        - resourcequotas
+        - replicationcontrollers
+        - limitranges
+        - persistentvolumeclaims
+        - persistentvolumes
+        - namespaces
+        - endpoints
+        - daemonsets
+        - deployments
+        - replicasets
+        - statefulsets
+        - cronjobs
+        - jobs
+        - horizontalpodautoscalers
+        - poddisruptionbudgets
+        - storageclasses
+        - volumeattachments
```

Comment:
can we remove this part? 
maybe we should add the logic in the kubernetes_state_core check, that if the list is empty or not define we use the default list of collector.
WDYT?

Reply:
> maybe we should add the logic in the kubernetes_state_core check, that if the list is empty or not define we use the default list of collector.

We already have this logic in the check itself, it would watch all the possible resources. The reason why we specify the collectors explicitly here is to make sure the helm chart knows what RBACs should be added to make the ksm core check work.

Reply:
@ahmed-mez 

but the RBAC list is managed by another file? 
should we build a function that generate the collectors list based on the rbac?

Reply:
The RBACs are defined here https://github.com/DataDog/helm-charts/blob/main/charts/datadog/templates/kube-state-metrics-core-rbac.yaml

> should we build a function that generate the collectors list based on the rbac?

not sure I understand, you mean we should expose the RBACs list in the values file? If this what you meant I think it's a good idea, it can also be useful for custom ksm core configurations.

Reply:
agree all this file can be removed

Reply:
Removed !